src/main/java/com/varukha/onlinebookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,33 @@
+package com.varukha.onlinebookstore.repository.book;
+
+import com.varukha.onlinebookstore.dto.BookSearchParametersDto;
+import com.varukha.onlinebookstore.model.Book;
+import com.varukha.onlinebookstore.repository.SpecificationBuilder;
+import com.varukha.onlinebookstore.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    public static final String TITLE_KEY = "title";
+    public static final String AUTHOR_KEY = "author";
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto searchParametersDto) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParametersDto.titles() != null
```

Comment:
you are duplicating the same code twice. for me it's a good candidate for a private method